synapse/compat.py

```diff
@@ -19,6 +19,7 @@
 version = (major, minor, micro)
 
 if version < (3, 0, 0):
+    import gzip
```

Comment:
`gzip` is imported for < 3.0.0 and >= 3.0.0, should we just import it at the top?